src/wallet/wallet.cpp

```diff
@@ -2700,12 +2700,20 @@ bool CWallet::CreateCoinStake(
     bool fKernelFound = false;
     int nAttempts = 0;
 
+    // update staker status (hash)
+    pStakerStatus->SetLastHash(pindexPrev->GetBlockHash());
+
     for (std::unique_ptr<CStakeInput>& stakeInput : listInputs) {
-        nCredit = 0;
+        //new block came in, move on
+        if (chainActive.Height() != pindexPrev->nHeight) return false;
+
         // Make sure the wallet is unlocked and shutdown hasn't been requested
-        if (IsLocked() || ShutdownRequested())
-            return false;
+        if (IsLocked() || ShutdownRequested()) return false;
+
+        // update staker status (time)
+        pStakerStatus->SetLastTime(GetCurrentTimeSlot());
```

Comment:
This should be updated with the `nTxNewTime` variable **after** the staking attempt (after the `Stake` method call independently if it was successful or not).

Reply:
It's not necessary. After the time is saved, the function goes to kernel (`Stake()`) right away.
`nTxNewTime` might result in being the subsequent time slot (in few cases), but that makes no difference to the logic of `CStakerStatus::IsActive` since it's checked against the *current* time.

Reply:
small possibility of ending up with two different slot times (one stored in the staking status and one used in the Stake process) and repeating the last slot hashing twice.

and well, don't see why we should call the `GetCurrentTimeSlot` twice, the Stake method is already doing it and the `nTxNewTime` is available when the Stake process returns.

Reply:
fair enough